### PR TITLE
coll: Use UT_array in tree and multicast operation

### PR DESCRIPTION
**Pull request by akhillanger**
This PR replaces `int * array` with `UT_array *array` at two places - for storing array of child ranks in tree, and for list of destination ranks in a multicast operation.

### CODE REVIEW

**Comment by akhillanger** [comment]
Tagging @raffenet for review.

**Comment by akhillanger** [comment]
test:jenkins/ch3/sock

File: src/mpi/coll/algorithms/treealgo/treeutil.c [hunk 1]
```diff
@@ -26,11 +26,7 @@ static int tree_add_child(MPII_Treealgo_tree_t * t, int rank)
     utarray_push_back(t->children, &rank, MPL_MEM_COLL);
     t->num_children++;
 
-  fn_exit:
```
Comment (raffenet) [inline src/mpi/coll/algorithms/treealgo/treeutil.c]: Unrelated changes. Probably another commit to remove unused labels would be good.

File: src/mpi/coll/transports/gentran/gentran_utils.c [hunk 2]
```diff
@@ -76,12 +76,11 @@ static void vtx_issue(int vtxid, MPII_Genutil_vtx_t * vtxp, MPII_Genutil_sched_t
 
             case MPII_GENUTIL_VTX_KIND__IMCAST:{
                     MPIR_Errflag_t errflag = MPIR_ERR_NONE;
-
```
Comment (raffenet) [inline src/mpi/coll/transports/gentran/gentran_utils.c]: Unrelated whitespace.

**Comment by raffenet** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most
